node/subscription.go

```diff
@@ -3,23 +3,20 @@ package node
 import (
 	"context"
 	"encoding/json"
-	"sync"
 
 	"github.com/pkg/errors"
 
 	"github.com/INFURA/go-ethlibs/jsonrpc"
 )
 
 type subscription struct {
-	response       *jsonrpc.RawResponse
-	subscriptionID string
-	ch             chan *jsonrpc.Notification
-
-	conn   Requester
-	ctx    context.Context
-	cancel context.CancelFunc
-	err    error
-	mu     sync.RWMutex
+	response        *jsonrpc.RawResponse
+	subscriptionID  string
+	notificationsCh chan *jsonrpc.Notification
+	dispatchCh      chan *jsonrpc.Notification
```

Comment:
I also considered naming these channels `internalCh` and `outputCh`, but opted for the current names since I called the internal method for queuing the notification `dispatch`.  If anyone has an opinion on which names are clearest please let me know.

Reply:
I think I'm missing something with the shutdown logic, but is StoppedCh needed? it looks like signalCh is the starting point that triggers stopping the subscription (through the close function) but is StoppedCh used other than being closed by signalCh ?

Reply:
Ya, `stoppedCh` is necessary since it's how `subscription.dispatch` and `subscription.stop` are allowed to exit early, as a channel that's closed always produces a zero value.  Without it, those two functions could hang forever as nothing should be reading `dispatchCh` or `signalCh` after the goroutine is shut down.  That said, I see I'm missing a return in the goroutine after the first read from `signalCh` so I'll add that and retest.

Reply:
Ok, cf6df09d088c9384a5923377d465cbdf504eea0a makes sure we return after closing.  I also used this as an opportunity to move the `close` calls into a defer rather than duplicating them in the two select statements.